Add keybindings for zoom in, zoom out and zoom reset

ArtemDrozdov opened the pull request:
Added keybindings for zoom in, zoom out and zoom reset according #1423 issue.

I checked zooming behavior in VSCode and they have next logic: default value is 0, minimum is -8 and maximum is 9. Zooming step is 1 which equals to 20%.

Short summary what was done in this PR:
1. Removed minimum value of 1.0 for zooming in ConfigurationStoreConnector;
2. Added keybindings to KeyBindingsStoreConnector for <kbd>Cmd</kbd> + <kbd>=</kbd>, <kbd>Ctrl</kbd> + <kbd>=</kbd>, <kbd>Cmd</kbd> + <kbd>-</kbd>, <kbd>Ctrl</kbd> + <kbd>-</kbd>, <kbd>Cmd</kbd> + <kbd>0</kbd>  and <kbd>Ctrl</kbd> + <kbd>0</kbd>;
3. Added Commands and zoom effect handler to VimStoreConnector.

Minimum value for zooming is 0 because on lower values there is no changes in UI.

And probably there should be some logic to process <kbd>Cmd</kbd> + <kbd>=</kbd> on Mac and <kbd>Ctrl</kbd> + <kbd>=</kbd> on Linux/Windows for example. Because with current implementation both of those combinations are working on Mac.

Would love to hear any comments and suggestion!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/onivim/oni2?pullRequest=1629) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/onivim/oni2?pullRequest=1629) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/onivim/oni2?pullRequest=1629) it.</sub>

glennsl commented on src/Store/VimStoreConnector.re:
```diff
@@ -975,6 +980,42 @@ let start =
     });
   };
 
+  let zoomStep = 0.2;
+  let defaultZoom = 1.0;
+  let minZoomValue = 0.0;
+  let maxZoomValue = 2.8;
```
Could you put these in a `Constants` sub-module at the top of the file?

ArtemDrozdov replied:
Yes, of course.

glennsl commented on src/Store/VimStoreConnector.re:
```diff
@@ -975,6 +980,42 @@ let start =
+      let currentZoomValue =
+        Core.Configuration.getValue(c => c.uiZoom, configuration);
+
+      let calculatedZoomValue =
+        switch (zoomAction) {
+        | ZoomIn => currentZoomValue +. zoomStep
+        | ZoomOut => currentZoomValue -. zoomStep
+        | ZoomReset => defaultZoom
+        };
```
What's the reason you pass in a `zoomAction` instead of just passing in the value directly?

ArtemDrozdov replied:
What do you mean by 
> instead of just passing in the value directly

Can't get your point.

glennsl replied:
Looking closer, I guess you can't pass in `currentZoomValue +. zoomStep)` directly instead of `ZoomIn` because you need `currentZoomValue`. But you could do `zoomEffect(state, zoom => zoom +. zoomStep)`.

glennsl commented on src/Store/VimStoreConnector.re:
```diff
@@ -975,6 +980,42 @@ let start =
+        | ZoomReset => defaultZoom
+        };
+
+      let newZoomValue =
+        calculatedZoomValue < minZoomValue
+          ? minZoomValue
+          : calculatedZoomValue > maxZoomValue
+              ? maxZoomValue : calculatedZoomValue;
```
You can use `Utility.IntEx.clamp` for this. It would be significantly more readable I think.

glennsl commented on src/Store/VimStoreConnector.re:
```diff
@@ -1032,6 +1073,18 @@ let start =
         state,
         synchronizeEditorEffect(state),
       )
+    | Command("workbench.action.zoomIn") => (
+        state,
+        zoomEffect(state, ZoomIn),
+      )
+    | Command("workbench.action.zoomOut") => (
+        state,
+        zoomEffect(state, ZoomOut),
+      )
+    | Command("workbench.action.zoomReset") => (
+        state,
+        zoomEffect(state, ZoomReset),
+      )
```
I don't see any reason for these and the effect to need to live in `VimStoreConnector`. Could you move it to `CommandStoreConnector` instead (unless there's a reason I don't see)?

ArtemDrozdov commented:
Pushed changes according @glennsl comments:

1. Moved effect logic from `VimStoreConnector` to `CommandStoreConnector`;
2. Added `Constants` submodule in `CommandStoreConnector`;
3. Changed effect logic a little bit.

glennsl commented:
Oh, CI fails on an unused variable warning. That should be a quick fix though.

glennsl commented on src/Store/CommandStoreConnector.re:
```diff
@@ -399,6 +433,18 @@ let start = (getState, contributedCommands) => {
+    ),
+    (
+      "workbench.action.zoomReset",
+      state => zoomEffect(state, zoom => Constants.defaultZoomValue),
```
```suggestion
      state => zoomEffect(state, _zoom => Constants.defaultZoomValue),
```
